modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -981,8 +996,9 @@ def _union_all(
         if (
             len(other_modin_frames) == 0
             or len(self.columns) == 0
-            or any((len(f.columns) != len(self.columns) for f in other_modin_frames))
-            or any((set(f._dtypes) != set(self._dtypes) for f in other_modin_frames))
+            or all(f._has_arrow_table() for f in ([self] + other_modin_frames))
```

Comment:
Why `_has_arrow_table` is used here?

Reply:
If all frames have an arrow table, we perform the arrow-based concatenation. There is a related issue - https://github.com/modin-project/modin/pull/5932.

Reply:
ok, thanks for the clarification!